File: python/binaryio.py
# coding=UTF-8
import struct

# Read string data from a binary file, stopping at the first null byte:
def readString(f):
	rtn = ''
	byte = f.read(1)
	while byte:
		unpacked = struct.unpack('B', byte)[0]

		if unpacked is 0:
			return rtn

		rtn = rtn + byte
		byte = f.read(1)

# Read a one byte int from the file:
def readByte(f):
	return struct.unpack('b', f.read(1))[0]

# Read a one byte unsigned int from the file:
def readUnsignedByte(f):
	return struct.unpack('B', f.read(1))[0]

# Read a two byte (short) int from the file:
def readInt(f):
	return struct.unpack('<h', f.read(2))[0]

# Read a four byte int from the file:
def readLong(f):
	return struct.unpack('<i', f.read(4))[0]

File: php/binaryio.php
<?php

class BinaryIO
{
	protected $file = null;
	protected $fileName = null;

	public function __construct($fileName)
	{
		$this->fileName = $fileName;
		$this->file		= fopen($fileName, 'rb');
	}

	public function readString()
	{
		$rtn  = '';
		$byte = fread($this->file, 1);

		while($byte && !feof($this->file))
		{
			$byte = unpack('a', $byte);
			$byte = array_shift($byte);

			if(empty($byte))
			{
				return $rtn;
			}

			$rtn .= $byte;
			$byte = fread($this->file, 1);
		}

		return $rtn;
	}

	public function readByte()
	{
		return array_shift(unpack('c', fread($this->file, 1)));
	}

	public function readUnsignedByte()
	{
		return array_shift(unpack('C', fread($this->file, 1)));
	}

	public function readInt()
	{
		return array_shift(unpack('s', fread($this->file, 2)));
	}

	public function readLong()
	{
		return array_shift(unpack('i*', fread($this->file, 4)));
	}
}